### DISCUSSION TITLE
Allocation of left and right tables in linker.predict

### DISCUSSION THREAD

**Discussion by JonathanLaidler** [moj-analytical-services/splink#1660] [Q&A] [answered]
Can I force which table becomes "_l" and which becomes "_r" in the results from linker.predict ?

The allocation of which input table becomes "_l" and which becomes "_r" does not necessarily correspond to the order in which I passed them when instantiating the linker. Is there a rule to this? e.g. the bigger table is always "_l"?
I would like to be confident in the allocation, so that I can, for example, filter on "unique_reference_r" when post-processing.

Thanks,

Jonny

**Comment by RobinL** [comment, marked as answer]
The join rule is that the concatenation of `source_dataset_l` and `unique_id_l` are always lower than `source_dataset_r` and `unique_id_r`, so so long as `source_dataset_l` is lower alphabetically than `source_dataset_r` it should end up on the left

That should happen automaticaly because the default aliases are `__splink__input_table_0` and `__splink__input_table_1`

If you're having trouble you could set input_table_aliases
`linker = DuckDBLinker([df_origin, df_destination], settings,input_table_aliases=["_a", "_b"])

Do you have example code where it's not working?

**Reply by JonathanLaidler** [reply]
Oh, they're alphabetical! I hadn't thought of this.
I was using the aliases but not realising that they influenced the order.

Thanks Robin!

**Reply by loukak** [reply]
That's useful to know Robin. I too assumed that the l and r suffixes in Splink output depend on the order you put the dataset names in the function. That would lead to post-processing errors as well. Using aliases sounds like a good idea.
